src/eth_plugin_handler.c

```diff
@@ -21,12 +21,8 @@ void eth_plugin_prepare_finalize(ethPluginFinalize_t *finalize) {
     memset((uint8_t *) finalize, 0, sizeof(ethPluginFinalize_t));
 }
 
-void eth_plugin_prepare_provide_token(ethPluginProvideToken_t *provideToken,
-                                      tokenDefinition_t *token1,
-                                      tokenDefinition_t *token2) {
+void eth_plugin_prepare_provide_token(ethPluginProvideToken_t *provideToken) {
     memset((uint8_t *) provideToken, 0, sizeof(ethPluginProvideToken_t));
-    provideToken->token1 = token1;
-    provideToken->token2 = token2;
```

Comment:
eth_plugin_prepare_provide_token is now called BEFORE getting token information. Hence the safe removal of those fields